### ISSUE TITLE
GM: Very slow acceleration when resuming from a stop or slow roll

### ISSUE THREAD

**Issue by Verylukyguy** [commaai/opendbc#1169]
### Describe the bug

Very slow acceleration when resuming from a stop or after slowing to a near stop for a lead car to turn; it does get normal at around 30 mph.

### What hardware does this issue affect?

comma two

### Which car does this affect?

GMC Acadia 2018

### Provide a route where the issue occurs

e38a02cae1ef1e0c|2022-04-15--07-43-35--3

### openpilot version

Master 0.8.14 @ 1fa1c44

### Additional info

This has been an ongoing issue from the beginning for the GM port.

@Twilsonco has fixed this in his fork and has even added a toggle for Normal/Sport/Eco/Creep acceleration modes.

**Comment by Verylukyguy** [comment]
@sshane Does merging #951 solve this?

**Comment by jyoung8607** [comment]
Should be resolved by #951.

**Comment by Verylukyguy** [comment]
I am still experiencing this issue, this did not resolve it.

**Comment by jyoung8607** [comment]
> I am still experiencing this issue, this did not resolve it.

Can you share a recent route on current openpilot?

**Comment by sshane** [comment]
As noted in that PR, it does not change any behavior. We still have to change the max value dependant on the car.

A recent route is a good idea.

**Comment by jyoung8607** [comment]
> As noted in that PR, it does not change any behavior. We still have to change the max value dependant on the car.

Ah, I knew the signal-width issue was sorted, but I missed that we weren't making use of the additional width yet.

@Verylukyguy We still do need routes demonstrating the issue as it stands.

**Comment by morrislee** [comment]
> > As noted in that PR, it does not change any behavior. We still have to change the max value dependant on the car.
> 
> Ah, I knew the signal-width issue was sorted, but I missed that we weren't making use of the additional width yet.
> 
> [@Verylukyguy](https://github.com/Verylukyguy) We still do need routes demonstrating the issue as it stands.

this is still happening for GM across the board, but as far as I know, more prominent on larger the vehicle as limits are currently set from smaller vehicles like the bolt. To fully implement fixes, FPv3 needs to be done first so we can identify the correct torque controller scaling from tire size, weight etc.

**Comment by Verylukyguy** [comment]
@jyoung8607 and @sshane I will be happy to provide all of the routes that I can to help resolve this.

My factory GM Full-Stop-and-Go ACC is not shy about accelerating and will be on and off of the accelerator and brake to keep a constant distance.

OpenPilot is lazy to resume from a start and accelerates slowly up to about 35 MPH before it starts to feel normal.

OpenPilot's braking is also late and harsh, which is uncomfortable and causes interventions and disengagements, but this must be a obviously separate issue.

**Comment by 1454** [comment]
This has, for the most part, been my experience. Even the changes mochi made helped some, but 0-35 is still pretty slow. Enough so that semis will drive away from me from a red light until after 35 when it starts catching back up.

**Comment by Verylukyguy** [comment]
@sshane There are three resumes from a stop in this route df51b2195de66a9b/00000005--739b7b6c5a/0
1.  df51b2195de66a9b/00000005--739b7b6c5a/2
2. df51b2195de66a9b/00000005--739b7b6c5a/5
3. df51b2195de66a9b/00000005--739b7b6c5a/6

This route has a few more:  df51b2195de66a9b/00000006--adf3ed521d/0

**Comment by Verylukyguy** [comment]
Can anyone look into this yet?
